### PR TITLE
Add User Progress Model

### PR DESCRIPTION
**Pull request by harshad00**
added  User Project Mode

### CODE REVIEW

**Comment by harshad00** [comment]
Hello, By mistake I wrote User Progress Model as User Project Model I am solving the problem can you check and give me feedback

File: src/constant/database.ts [hunk 1]
```diff
@@ -1,5 +1,7 @@
 const databaseModels = {
   PROJECT: 'Project',
+  PROGRESS_TASK: 'ProgressTask',
```
Comment (imsks) [inline src/constant/database.ts]: When naming a variable, make it descriptive like USER_PROJECT_PROGRESS

File: src/interfaces/database.ts [hunk 2]
```diff
@@ -40,6 +40,20 @@ export interface ProjectDocumentModel extends Document {
...
 
+
+export interface UserProgressTaskModel {
+  taskId: typeof Schema.Types.ObjectId;
+  isCompleted: boolean;
+  createdAt?: Date;
+  updatedAt?: Date;
+}
+
+export interface UserProgressModel {
```
Comment (imsks) [inline src/interfaces/database.ts]: While UserProgressModel is fine but we've different products which need User Progress Feature so we can't use generic UserProgressModel name we need to Identify Product name also like UserProjectProgressModel

**Review by imsks — CHANGES_REQUESTED** [review]

File: src/constant/database.ts (hunk 1 above)
Comment (harshad00) [reply src/constant/database.ts]: changed name

**Comment by imsks** [comment]
Implemented.

**Pull request closed without merging**